wrapper/main/src/main/kotlin/org/quiltmc/qkl/wrapper/main/math/Vec2fOps.kt

```diff
@@ -0,0 +1,123 @@
+package org.quiltmc.qkl.wrapper.main.math
+
+import net.minecraft.client.util.math.Vector2f
+import net.minecraft.util.math.Vec2f
+
+//region Standard math operators
+public operator fun Vec2f.plus(other: Vec2f): Vec2f {
+    return Vec2f(
+        this.x + other.x,
+        this.y + other.y
+    )
+}
+
+public operator fun Vec2f.plus(other: Float): Vec2f {
+    return Vec2f(
+        this.x + other,
+        this.y + other
+    )
+}
+
+public operator fun Vec2f.minus(other: Vec2f): Vec2f {
+    return Vec2f(
+        this.x - other.x,
+        this.y - other.y
+    )
+}
+
+public operator fun Vec2f.minus(other: Float): Vec2f {
+    return Vec2f(
+        this.x - other,
+        this.y - other
+    )
+}
+
+public operator fun Vec2f.times(other: Vec2f): Vec2f {
+    return Vec2f(
+        this.x * other.x,
+        this.y * other.y
+    )
+}
+
+public operator fun Vec2f.times(other: Float): Vec2f {
+    return Vec2f(
+        this.x * other,
+        this.y * other
+    )
+}
+
+public operator fun Vec2f.unaryMinus(): Vec2f {
+    return this.times(-1f)
+}
+//endregion
+
+//region xAssign math operators
+public operator fun Vec2f.plusAssign(other: Float) {
+    this.plus(other)
+}
+
+public operator fun Vec2f.plusAssign(other: Vec2f) {
+    this.plus(other)
+}
+
+public operator fun Vec2f.plusAssign(other: Vector2f) {
+    this.plus(other)
+}
+
+public operator fun Vec2f.minusAssign(other: Float) {
+    this.minus(other)
+}
+
+public operator fun Vec2f.minusAssign(other: Vec2f) {
+    this.minus(other)
+}
+
+public operator fun Vec2f.minusAssign(other: Vector2f) {
+    this.minus(other)
+}
+
+public operator fun Vec2f.timesAssign(other: Float) {
+    this.times(other)
+}
+
+public operator fun Vec2f.timesAssign(other: Vec2f) {
+    this.times(other)
+}
+
+public operator fun Vec2f.timesAssign(other: Vector2f) {
+    this.times(other)
+}
+//endregion
+
+//region Type compatibility operator variations
+public operator fun Vec2f.plus(other: Vector2f): Vec2f {
+    return Vec2f(
+        this.x + other.x,
+        this.y + other.y
+    )
+}
+
+public operator fun Vec2f.minus(other: Vector2f): Vec2f {
+    return Vec2f(
+        this.x - other.x,
+        this.y - other.y
+    )
+}
+
+public operator fun Vec2f.times(other: Vector2f): Vec2f {
+    return Vec2f(
+        this.x * other.x,
+        this.y * other.y
+    )
+}
+//endregion
+
+//region Vector specific operators
+public infix fun Vec2f.dot(other: Vec2f): Float {
```

Comment:
What about cross products and other operations (e.g. normalize, magnitude, etc.)?

Reply:
Two dimensional cross product ends up being a scalar instead so I ended up skipping it at the time, if it makes sense I can add it back though

Normalize could probably be implemented, but magnitude is just a property so I don't see much reason to add it